File: string/decrypt-string-from-alphabet-to-integer-mapping.py
# https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/
# tags - string
class Solution:
    def freqAlphabets(self, s: str) -> str:
        map = {}
        # ASCII code of 'a' = 97
        for i in range(26):
            key = str(i+1) if i < 9 else f'{i+1}#'
            map[key] = chr(i + 97)
        
        i = 0
        decoded = []
        while i < len(s):
            span = 1
            if i+2 < len(s) and s[i+2] == '#':
                span = 3
            decoded.append(map[s[i:i+span]])
            i += span
        return ''.join(decoded)

File: string/decrypt-string-from-alphabet-to-integer-mapping.js
// https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/
// tags - string
/**
 * @param {string} s
 * @return {string}
 */
var freqAlphabets = function (s) {
  let result = "";
  const CODE_A = "a".charCodeAt(0);
  for (let i = 0; i < s.length; ) {
    let chunk = s[i++];
    if (i + 1 < s.length && s[i + 1] === "#") {
      chunk += s[i];
      i += 2;
    }
    result += String.fromCharCode(CODE_A + Number(chunk) - 1);
  }
  return result;
};
